### PR TITLE
Docs: Intro to Storybook create addon section rework

### PR DESCRIPTION
**Pull request by jonniebigodes**
While reading "Intro to Storybook" and incorporating it with the "Create addons section" i was hitting a couple of issues with the present form of the document. 

What was done:
- Uniformization of the code base throughout the document.
- Rewording of the document to follow the style used with the rest of the documentation
-  Make it more inclusive, by removing some keywords like "easy" and "simple" in order to lower the barrier to newcomers.
- Added some screenshots to better illustrate certain steps, so that when the reader reaches a given point in the document it has something available to match his/hers work.

This a first draft and feel free to provide feedback

### CODE REVIEW

File: content/intro-to-storybook/react/en/creating-addons.md [hunk 1]
```diff
@@ -197,72 +249,66 @@ const Asset = ({ url }) => {
...
+If you take a closer look, you'll see that we're using the `styled` tag, this tag comes from the `@storybook/theming` package. Using this tag, will allow us to customize not only Storybook's theme but also the UI to our needs. Also [useStorybookState](https://storybook.js.org/docs/addons/api/#usestorybookstate), which is a real handy hook, that allows us to tap into Storybook's internal state so that we can fetch any bit of information present. In our case we're using it to fetch only the id of each story created.
 
-## Stateful addons
+### Displaying actual assets
 
-So 1 of our goals was:
+To actually see the assets displayed in our addon, we need to copy them over to the `public` folder and adjust the `addParameter` option to reflect these changes.
 
-> - should support multiple assets, just in case there will be multiple versions or themes
+Storybook will pick on the change and will load the assets, but for now, only the first one.
```
Comment (domyen) [inline content/intro-to-storybook/react/en/creating-addons.md]: "will pick up"

File: content/intro-to-storybook/react/en/creating-addons.md [hunk 2]
```diff
@@ -197,72 +249,66 @@ const Asset = ({ url }) => {
...
+- ✔️ Display the design asset in a panel
+- ✔️ Support images, but also urls for embedding
+- ❌ Should support multiple assets, just in case there will be multiple versions or themes
 
-## Using storybook UI inside your custom addon
+We're almost there, only one goal remaining.
 
-Now we have state, but no way of changing the state.
-We could of course create our own UI, but let's instead use some UI that storybook has already made.
+For the final one, we're going to need some sort of state, we could use React's `useState`, or if we were working with class components `this.setState()`. But instead we're going to use Storybook's own `useAddonState`, which gives us a means to persist the addon state, and avoid creating extra logic to persist the local state. Also use another UI element from Storybook, the `ActionBar`, which will allow us to change between items.
```
Comment (domyen) [inline content/intro-to-storybook/react/en/creating-addons.md]: "We'll also use another UI element from Storybook..."

File: content/intro-to-storybook/react/en/creating-addons.md [hunk 3]
```diff
@@ -376,37 +395,46 @@ export const Content = () => {
...
+</details>
 
-You can open a PR on the storybook documentation to have your addon featured there.
+## Next steps
 
-## What's next?
+The next logical step for our addon, would be to make it it's own package and allow it to be distributed with your team and possibly with the rest of the community.
 
-This is an example of an addon that uses parameters and displays it in a panel, but that's just one of the many options you have.
+But that's beyond the scope of this tutorial. This example demonstrates how you can use the Storybook API to create your own custom addon to further enchance your development workflow.
```
Comment (domyen) [inline content/intro-to-storybook/react/en/creating-addons.md]: "enhance"

File: content/intro-to-storybook/react/en/creating-addons.md [hunk 4]
```diff
@@ -376,37 +395,46 @@ export const Content = () => {
...
 
-With these principles you're able to display you custom UI in a variety of places in the storybook UI.
+Not only what we've accomplished here, but additionally you can also:
```
Comment (domyen) [inline content/intro-to-storybook/react/en/creating-addons.md]: "Learn how to further customize your addon:"

File: content/intro-to-storybook/react/en/creating-addons.md [hunk 5]
```diff
@@ -417,16 +445,16 @@ You can:
 
 And much more!
 
-# Dev Kits
+<div class="aside">Should you create a new addon and you're interested in having it featured, feel free to open a PR in the Storybook documentation to have it featured.</div>
+
+### Dev kits
 
-To help you write addons, the Storybook team has developed `dev-kits`. 
+To help you jumpstart the addon development, the Storybook teams has developed some `dev-kits`.
```
Comment (domyen) [inline content/intro-to-storybook/react/en/creating-addons.md]: teams --> team

File: content/intro-to-storybook/react/en/creating-addons.md [hunk 6]
```diff
@@ -376,37 +395,46 @@ export const Content = () => {
...
+The next logical step for our addon, would be to make it it's own package and allow it to be distributed with your team and possibly with the rest of the community.
 
-This is an example of an addon that uses parameters and displays it in a panel, but that's just one of the many options you have.
+But that's beyond the scope of this tutorial. This example demonstrates how you can use the Storybook API to create your own custom addon to further enchance your development workflow.
 
-With these principles you're able to display you custom UI in a variety of places in the storybook UI.
+Not only what we've accomplished here, but additionally you can also:
 
-You can:
 - [add buttons in the storybook toolbar](https://github.com/storybookjs/storybook/blob/next/addons/viewport/src/register.tsx#L8-L15)
```
Comment (domyen) [inline content/intro-to-storybook/react/en/creating-addons.md]: Capitalize Storybook

File: content/intro-to-storybook/react/en/creating-addons.md [hunk 7]
```diff
@@ -417,16 +445,16 @@ You can:
...
+<div class="aside">Should you create a new addon and you're interested in having it featured, feel free to open a PR in the Storybook documentation to have it featured.</div>
+
+### Dev kits
 
-To help you write addons, the Storybook team has developed `dev-kits`. 
+To help you jumpstart the addon development, the Storybook teams has developed some `dev-kits`.
 
-These packages are like little starter-sets for you to start developing your own addon.
-The addon we just created is based on the `addon-parameters` dev-kit.
+These packages work like little starter-sets, for you to start developing your own custom addons.
```
Comment (domyen) [inline content/intro-to-storybook/react/en/creating-addons.md]: These packages are starter-kits to help you start building your own addons.

**Comment by domyen** [comment]
You're a **legend** @jonniebigodes! 🙏


**Added comments above, some feedback**
- This image doesn't appear to resolve on the deploy preview.
image
- Can we remove the commit hash from the frontmatter of this doc. I realize that it's linked to the previous chapter (and we don't have a commit has for this chapter).

**Comment by jonniebigodes** [comment]
@domyen i'll address the changes during the weekend. Also regarding the missing image, it was something that i would like to address. I thought about it and i was wondering where i could find some stock images to use. That are related to Storybook, so that they can be used safely without me using a random image, so that the document aligns itself with the rest of the present documentation.

Feel free to provide feedback

File: content/intro-to-storybook/react/en/creating-addons.md (hunk 4 above)
Comment (domyen) [reply content/intro-to-storybook/react/en/creating-addons.md]: I meant replace "Not only what we've accomplished here, but additionally you can also:" with "Learn how to further customize your addon:"

Comment (jonniebigodes) [reply content/intro-to-storybook/react/en/creating-addons.md]: i read it wrong, Sorry about that. I'll fix it and commit the change shortly.

**Comment by domyen** [comment]
Nice this is ready to merge pending merge conflicts 👍

**Comment by jonniebigodes** [comment]
Merge conflicts done. Think that's ready to merge. I'll adjust the image tomorrow and make a smaller pull request to address it. Sounds good?

**Comment by domyen** [comment]
Yup sgtm

**Pull request merged**